src/lib/y2users/linux/users_writer.rb

```diff
@@ -0,0 +1,366 @@
+      #
+      # @param config [Config] see #config
+      # @param initial_config [Config] see #initial_config
+      def initialize(config, initial_config)
+        textdomain "users"
+
+        @config = config
+        @initial_config = initial_config
+      end
+
+      # Creates the new users
+      #
+      # @param issues [Y2Issues::List] the list of issues found while writing changes
+      def add_users(issues)
+        new_users = config.users.without(initial_config.users.ids)
+
+        new_users.each { |u| add_user(u, issues) }
+      end
+
+      # Applies changes for the edited users
+      #
+      # @param issues [Y2Issues::List] the list of issues found while writing changes
+      def edit_users(issues)
+        edited_users = config.users.changed_from(initial_config.users)
+
+        edited_users.each do |user|
+          initial_user = initial_config.users.by_id(user.id)
+          edit_user(user, initial_user, issues)
+        end
+      end
+
+    private
+
+      # Configuration containing the users and groups that should exist in the system after writing
+      #
+      # @return [Config]
+      attr_reader :config
+
+      # Initial state of the system (usually a Y2Users::Config.system in a running system) that will
+      # be compared with {#config} to know what changes need to be performed.
+      #
+      # @return [Config]
+      attr_reader :initial_config
+
+      # Default values to use during user creation
+      #
+      # @return [UserDefaults]
+      def defaults
+        config.user_defaults
+      end
+
+      # Command for creating new users
+      USERADD = "/usr/sbin/useradd".freeze
+      private_constant :USERADD
+
+      # Command for modifying users
+      USERMOD = "/usr/sbin/usermod".freeze
+      private_constant :USERMOD
+
+      # Command for setting a user password
+      #
+      # This command is "preferred" over
+      #   * the `passwd` command because the password at this point is already
+      #   encrypted (see Y2Users::Password#value). Additionally, this command
+      #   requires to enter the password twice, which it's not possible using
+      #   the Cheetah stdin argument.
+      #
+      #   * the `--password` useradd option because the encrypted
+      #   password is visible as part of the process name
+      CHPASSWD = "/usr/sbin/chpasswd".freeze
+      private_constant :CHPASSWD
+
+      # Command for configuring the attributes in /etc/shadow
+      CHAGE = "/usr/bin/chage".freeze
+      private_constant :CHAGE
+
+      # Executes the command for creating the user
+      #
+      # @param user [User] the user to be created on the system
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def add_user(user, issues)
+        Yast::Execute.on_target!(USERADD, *useradd_options(user))
+        change_password(user, issues) if user.password
+        write_auth_keys(user, issues)
+      rescue Cheetah::ExecutionFailed => e
```

Comment:
I have to say that this desing kind of confuse me as I thought that it catched execute failure also for chpasswd and others. I think it kind of hit by issue that it is different level of abstraction. Maybe remove this rescue completely and separate Yast::Execute to own method with own rescue? And here have just
```ruby
call_useradd(user, issues)
change_password(user, issues) if user.password
write_auth_keys(user, issues)
```

and when we are at it I see also that if inconsistent and probably should also live in change_password method.

Reply:
Again, this is not new code. Just moved to another file. Let's stay focused on this PR by not re-reviewing that. What about a separate card for this kind of things we want to revisit?